R/Rules-methods.R

```diff
@@ -1001,6 +1001,304 @@ setMethod(
+        colour = "black"
+      ) +
+      geom_text(
+        data = data.frame(x = as.factor(dose_target)),
+        aes(.data$x, 0),
+        label = "Est",
+        vjust = -0.5,
+        hjust = -0.5,
+        colour = "black",
+        angle = 90
+      ) +
+      xlab("Dose") +
+      ylab(paste("Allocation criterion [%]"))
+
+    if (nextBest@method == "max") {
+      p <- p +
+        geom_text(
+          mapping = aes(
+            x = Inf,
+            y = Inf
+          ),
+          label = paste("Method:", nextBest@method),
+          hjust = 1,
+          vjust = 1
+        )
+    }
+
+    if (is.finite(doselimit)) {
+      p <- p +
+        geom_vline(
+          xintercept = doselimit_factor,
+          colour = "red", lwd = 1.1
+        ) +
+        geom_text(
+          data = data.frame(x = doselimit_factor),
+          aes(.data$x, 0),
+          label = "Max",
+          vjust = -0.5,
+          hjust = -1.5,
+          colour = "red",
+          angle = 90
+        )
+    }
+
+    p <- p +
+      geom_vline(
+        xintercept = as.factor(next_dose),
+        colour = "blue", lwd = 1.1
+      ) +
+      geom_text(
+        data = data.frame(x = as.factor(next_dose)),
+        aes(.data$x, 0),
+        label = "Next",
+        vjust = -0.5,
+        hjust = -2.5,
+        colour = "blue",
+        angle = 90
+      )
+
+    list(
+      value = next_dose,
+      allocation = cbind(dose = allocation_crit_dose, allocation = allocation_crit),
+      plot = p
+    )
+  })
+
+## NextBestProbMTDMinDist ----
+
+#' @describeIn nextBest find the next best dose based with the highest
+#'  probability of having a toxicity rate with minimum distance to the
+#'  target toxicity level.
+#'
+#' @aliases nextBest-NextBestProbMTDMinDist
+#'
+#' @export
+#' @example examples/Rules-method-nextBest-NextBestProbMTDMinDist.R
+#'
+setMethod(
+  f = "nextBest",
+  signature = signature(
+    nextBest = "NextBestProbMTDMinDist",
+    doselimit = "numeric",
+    samples = "Samples",
+    model = "GeneralModel",
+    data = "Data"
+  ),
+  definition = function(nextBest, doselimit, samples, model, data, ...) {
+
+    # Matrix with samples from the dose-tox curve at the dose grid points.
+    prob_samples <- sapply(data@doseGrid, prob, model = model, samples = samples)
+
+    # Determine which dose level has the minimum distance to target.
+    dose_min_mtd_dist <- apply(prob_samples, 1,
+                               function(x) which.min(abs(x - nextBest@target)))
+    prob_mtd_min_dist <- sapply(1:data@nGrid,
+                                function(x) mean(dose_min_mtd_dist == x))
+    allocation_crit <- prob_mtd_min_dist
+    names(allocation_crit) <- as.character(data@doseGrid)
+
+    # In case that placebo is used, placebo and the portion that is not assigned
+    # to any dose of the grid are merged.
+    if (data@placebo) {
+      allocation_crit[2] <- sum(allocation_crit[1:2])
+      allocation_crit <- allocation_crit[-1]
+    }
```

Comment:
- to 1: You are right, but I think the comment is ok. The idea is that the prob that is assigned to placebo and no dose safe are summed up, so that there is only one portion that is outside of the dose grid (increasing the assessment that there may be no safe dose).
to 2: I think it can never happen here. The `allocation_crit` is calculated for `0:data@nGrid` and if placebo is used, there must be at least 2 elements in `allocation_crit`, but not necessarily in `data@nGrid` (if it is possible to create Data with placebo only).

- I am not sure who table can be used to sum the first two elements of a vector into one new element (I will further think about this)

Reply:
1. You are right, the comment is misleading. In this method there is no portion outside the grid, as smallest distance to target can always be calculated. I will change the comment to:
>     # In case that placebo is used, placebo and the first non-placebo dose
>     # of the grid are merged.

2. I think it is not possible to create a pure placebo dataset:
> Error in validObject(.Object) :
> invalid class “Data” object: A cohort with only placebo is not allowed

I.e. in case that placebo is used, the vector contains at least two elements.

Reply:
good thanks